multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/PollServiceKeyDeletionLastOperationExecution.java

```diff
@@ -0,0 +1,14 @@
+package org.cloudfoundry.multiapps.controller.process.steps;
+
+import com.sap.cloudfoundry.client.facade.domain.CloudServiceKey;
+
+public class PollServiceKeyDeletionLastOperationExecution extends PollServiceKeyLastOperationExecution {
+
+    @Override
+    protected AsyncExecutionState checkServiceKeyLastOperation(CloudServiceKey serviceKey, ProcessContext context) {
+        if (serviceKey == null) {
```

Comment:
This class is used only during the polling of a deletion. If the key gets deleted it will return null here, this is why this check exists. It cannot be null in the parent as it is used for creation